vulnerabilities/data_dump.py

```diff
@@ -190,3 +190,44 @@ def npm_dump(extract_data):
                     vulnerability=vulnerability,
                     package=package_fixed
                 )
+
+
+def ruby_dump(extract_data):
+    for package_data in extract_data:
+
+        if package_data['severity']:
+            vulnerability, _ = Vulnerability.objects.get_or_create(
+                cve_id=package_data['cve_id'],
+                summary=package_data['summary'],
+                cvss=package_data['severity']
+            )
+        else:
+            vulnerability, _ = Vulnerability.objects.get_or_create(
+                cve_id=package_data['cve_id'],
+                summary=package_data['summary']
+            )
+
+        VulnerabilityReference.objects.get_or_create(
+            vulnerability=vulnerability,
+            url=package_data['advisory']
+        )
+
+        for version in package_data['affected_versions']:
+            affected_package = Package.objects.create(
```

Comment:
We also need `type="gem"` here to enable querying by "Package URL".

However, this is only correct for actual Ruby gems. The data we are storing here also contains vulnerabilities in Ruby implementations (MRI (aka CRuby), JRuby and Rubinius (aka rbx)). Those are not gems and therefore not distributed via rubygems.org, as far as I can tell. Arguably those are not even packages. They are the stuff that goes into packages distributed by Debian, Fedora, etc. This issue has come up in https://github.com/package-url/purl-spec/issues/13 a long time ago and I don't think it will be solved before we merge this PR. So for the time being, I suggest we skip security advisories for Ruby implementations.